stl/inc/queue

```diff
@@ -108,18 +108,18 @@ public:
         _Swap_adl(c, _Right.c);
     }
 
-    template <class _Ty, class _Container>
-    friend bool operator==(const queue<_Ty, _Container>&, const queue<_Ty, _Container>&);
-    template <class _Ty, class _Container>
-    friend bool operator!=(const queue<_Ty, _Container>&, const queue<_Ty, _Container>&);
-    template <class _Ty, class _Container>
-    friend bool operator<(const queue<_Ty, _Container>&, const queue<_Ty, _Container>&);
-    template <class _Ty, class _Container>
-    friend bool operator>(const queue<_Ty, _Container>&, const queue<_Ty, _Container>&);
-    template <class _Ty, class _Container>
-    friend bool operator<=(const queue<_Ty, _Container>&, const queue<_Ty, _Container>&);
-    template <class _Ty, class _Container>
-    friend bool operator>=(const queue<_Ty, _Container>&, const queue<_Ty, _Container>&);
+    template <class _Ty2, class _Container2>

```

Comment:
Clang emits warnings about these shadowing the class template parameters `_Ty` and `_Container`.

Reply:
Why do they even need the template parameters at all? Wouldnt it be possible to just do 
```cpp
friend bool operator>=(const queue&, const queue&);
```

Or would that require to move the implementation into the class. I remember an interesting [talk](https://www.youtube.com/watch?v=POa_V15je8Y) about making friends from Dan Saks

Reply:
Making these operators hidden friends is interesting, but user-observable and therefore outside the scope of this PR. I *do* think it would be cleaner to befriend the corresponding specializations of these operator templates rather than all specializations, however, which also removes the need for template parameters here. I'll do that insteead.

Reply:
:+1: